src/emailmessagelistmodel.cpp

```diff
@@ -305,6 +306,13 @@ QVariant EmailMessageListModel::data(const QModelIndex & index, int role) const
     else if (role == MessageSenderFirstCharRole) {
         QString sender = messageMetaData.from().name();
         return firstChar(sender);
+    } else if (role == MessageParsedSubject) {
+        // Filter <img> and <ahref> html tags to make the text suitable to be displayed in a qml
+        // label using StyledText(allows only small subset of html)
+        QString subject = data(index, QMailMessageModelBase::MessageSubjectTextRole).toString();
+        subject.replace(QRegExp("<\\s*img", Qt::CaseInsensitive), "<no-img");
+        subject.replace(QRegExp("<\\s*a\\s*href", Qt::CaseInsensitive), "<no-href");
```

Comment:
This tag is <a> with a 'href' attribute - there's no reason to require that href would be the first attribute.  I would simply suppress all anchor tags.

    subject.replace(QRegExp("<\\s*a\b", Qt::CaseInsensitive), "<no-a");